patches/components-version_info-BUILD.gn.patch

```diff
@@ -0,0 +1,18 @@
+diff --git a/components/version_info/BUILD.gn b/components/version_info/BUILD.gn
+index 8fbad36091b4a1e1baea40911599ac0d2d12aa0a..7d36d11a23e54e9e8eac22d8e9ae0c4a402eb367 100644
+--- a/components/version_info/BUILD.gn
++++ b/components/version_info/BUILD.gn
+@@ -21,6 +21,13 @@ static_library("version_info") {
+     "//base",
+   ]
+ 
++  if (brave_chromium_build) {
```

Comment:
I think we can avoid this by adding a method to our own brave version_info_values. Then the generated header can go inside browser/version_info_values.cc and this dep is no longer needed

Reply:
Then, how this `version_info` target rebuild is triggered when `brave-browser/package.json` file is changed?

Reply:
it doesn't have to, it becomes a link-time dependency

Reply:
this dep will move to the target containing `browser/version_info_values.cc`

Reply:
I think it should be rebuild because we override `GetProductNameAndVersionForUserAgent()` in `chromium_src` and it uses `BRAVE_CHROMIUM_VERSION`. Am I wrong?

Reply:
the whole point is not to use BRAVE_CHROMIUM_VERSION directly. You wrap that inside a method in a non-generated header so this target gets decoupled from the generated header dependency

Reply:
`#include "brave/version.h"` moves to `browser/version_info_values.cc` and it becomes strictly a link-time dep

Reply:
If I add `GetChromiumVersionString()` in `brave/version_info_values.h` and it is used by `GetProductNameAndVersionForUserAgent()`, how this `version_info` target knows the changing of `brave/version_info_values.cc/h`?

Reply:
Ah.. I see!

Reply:
to make it even simpler, just create a new `brave/chromium_version_info.cc/h` file with just that one method. Nothing will ever change in the header file so components version_info_values never has to worry about it. Changes to `chromium_version_info.cc` are irrelevant because they don't impact anything in components/version_info_values

Reply:
the only danger in the original method I proposed is that we might change something else in `browser/version_info_values.h` which would break at runtime, but this eliminates that problem as well

Reply:
I tried with additional api in `brave/...`.
Then, this causes a lot of dependency modification especially for component build.(also release build)
For that, I should add new dependency to dll targets that depends on `version_info`.
This will cause more patch file creation than this. WDYT?

Reply:
I'll take a look at it because that shouldn't be the case. I think maybe you're missing something here

Reply:
you're putting those in the `.cc` file right, not the header file? The only output of this is the header file so it doesn't make sense that changing what I suggested would create new dependencies. All it should do is remove this one